src/coreclr/jit/gentree.cpp

```diff
@@ -23446,6 +23447,29 @@ GenTree* Compiler::gtNewSimdShuffleNode(
             // swap the operands to match the encoding requirements
             retNode = gtNewSimdHWIntrinsicNode(type, op2, op1, NI_AVX2_PermuteVar8x32, simdBaseJitType, simdSize);
         }
+        else if (elementSize == 2)
+        {
```

Comment:
It might be worth an `assert(compSupportsIsaDebugOnly(...)` here and for the other new path. To help ensure its covered across a refactoring, particularly since its not an obvious connection to the original check above.

Reply:
Done